components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaProducer.java

```diff
@@ -0,0 +1,168 @@
+        OutputStream result = new ByteArrayOutputStream();
+        ContentHandler contentHandler = getContentHandler(this.tikaConfiguration, result);
+        ParseContext context = new ParseContext();
+        context.set(Parser.class, this.parser);
+        Metadata metadata = new Metadata();
+        this.parser.parse(inputStream, contentHandler, metadata, context);
+        convertMetadataToHeaders(metadata, exchange);
+        return result;
+    }
+
+    private void convertMetadataToHeaders(Metadata metadata, Exchange exchange) {
+        if (metadata != null) {
+            for (String metaname : metadata.names()) {
+                exchange.getIn().setHeader("Tika" + metaname, metadata.get(metaname));
+            }
+        }
+    }
+
+    private ContentHandler getContentHandler(TikaConfiguration configuration, OutputStream outputStream)
+            throws TransformerConfigurationException, UnsupportedEncodingException {
+
+        ContentHandler result = null;
+
+        TikaParseOutputFormat outputFormat = configuration.getTikaParseOutputFormat();
+        String encoding = Charset.defaultCharset().name();
+        switch (outputFormat) {
+        case xml:
+            result = getTransformerHandler(outputStream, "xml", encoding, true);
+            break;
+        case text:
+            result = new BodyContentHandler(outputStream);
+            break;
+        case textMain:
+            result = new BoilerpipeContentHandler(getOutputWriter(outputStream, encoding));
+            break;
+        case html:
+            result = new ExpandedTitleContentHandler(getTransformerHandler(outputStream, "html", encoding, true));
+            break;
+        default:
+            throw new IllegalArgumentException(
+                    String.format("Unknown format %s", tikaConfiguration.getTikaParseOutputFormat()));
+        }
+        return result;
+    }
+
+    private TransformerHandler getTransformerHandler(OutputStream output, String method, String encoding,
+            boolean prettyPrint) throws TransformerConfigurationException {
+        SAXTransformerFactory factory = (SAXTransformerFactory) SAXTransformerFactory.newInstance();
+        TransformerHandler handler = factory.newTransformerHandler();
+        handler.getTransformer().setOutputProperty(OutputKeys.METHOD, method);
+        handler.getTransformer().setOutputProperty(OutputKeys.INDENT, prettyPrint ? "yes" : "no");
+        if (encoding != null) {
+            handler.getTransformer().setOutputProperty(OutputKeys.ENCODING, encoding);
+        }
+        handler.setResult(new StreamResult(output));
+        return handler;
+    }
+
+    private Writer getOutputWriter(OutputStream output, String encoding) throws UnsupportedEncodingException {
+        if (encoding != null) {
+            return new OutputStreamWriter(output, encoding);
+        } else if (System.getProperty("os.name").toLowerCase(Locale.ROOT).startsWith("mac os x")) {

```

Comment:
This is perhaps something unique to the TikaCLI that I pulled from here https://github.com/apache/tika/commit/ac74696e322fa93bdbec982f55226f717be66d24 .  I'm thinking perhaps the better option might be just to allow the developer to explicitly set the output content encoding via the url as follows: 
tika:parse?tikaParseOutputEncoding=UTF-8

I think that might be preferable than doing something that the developer might not be aware of.